### PR TITLE
tune error code and message of Errors.cs

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#372807](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/372807)

### CODE REVIEW

File: test/docfx.Test/lib/MonikerRangeParserTest.cs [hunk 1]
```diff
@@ -117,14 +117,14 @@ public void TestEvaluateMonikerRange(string rangeString, string expectedMonikers
...
         [Theory]
-        [InlineData("netcore-xp", "Invalid moniker range 'netcore-xp': Moniker 'netcore-xp' is not defined.")]
-        [InlineData("netcore-1.0 < || netcore-2.0", "Expect a moniker string, but got ' || netcore-2.0'.")]
-        [InlineData(">netcore&-1.0", "Parse ends before reaching end of string, unrecognized string: '&-1.0'.")]
-        [InlineData(">=>netcore&-1.0", "Expect a moniker string, but got '>netcore&-1.0'.")]
-        [InlineData(">netcore<-1.0", "Invalid moniker range '>netcore<-1.0': Moniker 'netcore' is not defined.")]
-        [InlineData(">netcore<-1.0 ||| >netcore-2.0", "Expect a comparator set, but got '| >netcore-2.0'.")]
-        [InlineData(">netcore<-1.0 || ||", "Expect a comparator set, but got ' ||'.")]
-        [InlineData(">netcore<-1.0 || <", "Expect a moniker string, but got ''.")]
+        [InlineData("netcore-xp", "Invalid moniker range: 'netcore-xp'. Moniker 'netcore-xp' is not defined.")]
```
Comment (yufeih) [inline test/docfx.Test/lib/MonikerRangeParserTest.cs:120]:
```suggestion
        [InlineData("netcore-xp", "Invalid moniker range: Moniker 'netcore-xp' is not defined.")]
```

**Review by yufeih — APPROVED** [review]

File: test/docfx.Test/lib/MonikerRangeParserTest.cs (hunk 1 above)
Comment (fkpwolf) [reply test/docfx.Test/lib/MonikerRangeParserTest.cs:120]:
```
        [InlineData("netcore-xp", "Invalid moniker range: 'netcore-xp'. Moniker 'netcore-xp' is not defined.")]
        [InlineData(">netcore<-1.0", "Invalid moniker range: '>netcore<-1.0'. Moniker 'netcore' is not defined.")]
```
If I changed first line, then second line will be 
```
[InlineData(">netcore<-1.0", "Invalid moniker range: Moniker 'netcore' is not defined.")]
```
They uses same string template. Is this error message accepted?

Comment (yufeih) [reply test/docfx.Test/lib/MonikerRangeParserTest.cs:120]: How about change the MonikerRangeInvalid message template to `Invalid moniker range '{monikerRange}': {message}`?

Comment (fkpwolf) [reply test/docfx.Test/lib/MonikerRangeParserTest.cs:120]: some callers have no `monikerRange` parameter. If so, 
```
[InlineData(">netcore<-1.0 ||| >netcore-2.0", "Invalid moniker range: Expect a comparator set, but got '| >netcore-2.0'.")]
```
will become into:
```
[InlineData(">netcore<-1.0 ||| >netcore-2.0", "Invalid moniker range '| >netcore-2.0': Expect a comparator set, but got '| >netcore-2.0'.")]
```
Does this make sense?

**Comment by herohua** [comment]
@928PJY Can you also review this PR?

File: src/docfx/lib/moniker/EvaluatorWithMonikersVisitor.cs [hunk 2]
```diff
@@ -26,7 +26,8 @@ public EvaluatorWithMonikersVisitor(MonikerDefinitionModel monikerDefinition)
         {
             if (!MonikerOrder.TryGetValue(expression.Operand, out var moniker))
             {
-                return (Errors.Versioning.MonikerRangeMissing(monikerRange, expression.Operand), Array.Empty<Moniker>());
+                return (Errors.Versioning.MonikerRangeInvalid(
+                    monikerRange, $"'{monikerRange}'. Moniker '{expression.Operand}' is not defined."), Array.Empty<Moniker>());
```
Comment (yufeih) [inline src/docfx/lib/moniker/EvaluatorWithMonikersVisitor.cs:30]:
```suggestion
                    monikerRange, $"Moniker '{expression.Operand}' is not defined for range '{monikerRange}'.."), Array.Empty<Moniker>());
```

**Pull request merged**